apps/extension/src/contents/inpage.ts

```diff
@@ -48,8 +48,11 @@ export const windowAI: WindowAI<ModelID> = {
     return new Promise((resolve, reject) => {
       _addResponseListener<CompletionResponse>(requestId, (res) => {
         if (isOk(res)) {
-          resolve(shouldReturnMultiple ? res.data : res.data[0])
-          onStreamResult && onStreamResult(res.data[0], null)
+          if (!res.data[0].isPartial) {
+            resolve(shouldReturnMultiple ? res.data : res.data[0])
+          } else {
```

Comment:
We might not need the else here, since a stream listener might want to check if their stream is done by checking the isPartial for each callback as well.

Reply:
I think that would break all current apps, since they're concatenating each new stream result to the previous ones.

I think streamers can just check for completion by handling the resolution of the surrounding promise

Reply:
Ah I got confused by the use of res.data for both call - thought they were similar.